src/olympia/devhub/views.py

```diff
@@ -1490,32 +1490,33 @@ def submit_finish(request, addon_id, addon):
     # Bounce to the versions page if they don't have any versions.
     if not addon.versions.exists():
         return redirect(addon.get_dev_url('versions'))
-    sp = addon.current_version.supported_platforms
-    is_platform_specific = sp != [amo.PLATFORM_ALL]
+    uploaded_version = addon.versions.latest()
```

Comment:
Something about using `versions.latest` rather than `addons.find_latest_version` makes me nervous, but I can't think of a scenario where its a problem....